modal/verify-nft.js

```diff
@@ -79,42 +86,66 @@ module.exports = {
               await interaction.member.roles.add(role)
               addedRoles.push(roleEmbed(interaction, role.name))
               // Everything has been allocated, lets upsert into the UserInscriptions table
-              await UserInscriptions.upsert({
-                userId: interaction.user.id,
-                inscriptionId: inscription.id,
+              const userInscription = await UserInscriptions.findOne({
+                where: {
+                  inscriptionId: inscription.id,
+                  userId: interaction.user.id,
+                },
               })
+              if (!userInscription) {
+                await UserInscriptions.create({
+                  inscriptionId: inscription.id,
+                  userId: interaction.user.id,
+                })
+              }
             } else {
               notFoundRoles.push(role.name)
             }
           }
         }
 
-        // Valid roles
-        if (addedRoles.length > 0) {
-          const roleRef = addedRoles.length > 1 ? 'roles' : 'role'
-          const embed = successEmbed(
-            'Successfully verified',
-            `Your signature was validated and you were assigned the ${addedRoles.join(' ')} ${roleRef}.`
-          )
-
-          return interaction.editReply({ embeds: [embed], ephemeral: true })
-        }
-
-        // Invalid Roles
-        if (notFoundRoles.length > 0) {
-          const embed =
-            notFoundRoles.length > 1
-              ? warningEmbed('Roles not found', `The **${notFoundRoles.join(' ')}** roles aren't available.`)
-              : warningEmbed('Role not found', `The **${notFoundRoles.join(' ')}** role isn't available.`)
-          return interaction.editReply({ embeds: [embed], ephemeral: true })
-        }
+        const collections = await Collections.findAll({
+          where: {
+            channelId: interaction.channelId,
+          },
+          attributes: [
+            'id',
+            'name',
+            'role',
+            [sequelize.fn('COUNT', sequelize.col('Inscriptions.id')), 'inscriptionCount'],
+          ],
+          include: {
+            model: Inscriptions,
+            include: {
+              model: UserInscriptions,
+              where: {
+                userId: interaction.user.id,
+              },
+            },
+          },
+          having: {
+            inscriptionCount: {
+              [Op.gt]: 0,
+            },
+          },
+          group: ['Collections.id'],
+        })
 
-        // Catch where no collections were matched
-        const warning = warningEmbed(
```

Comment:
Should we not continue to catch where no inscriptions were matched?

Reply:
Sorry, this is my mistake.

I will add it soon